plumbing/cache/object_lru.go

```diff
@@ -67,6 +72,9 @@ func (c *ObjectLRU) Put(obj plumbing.EncodedObject) {
 // Get returns an object by its hash. It marks the object as used. If the object
 // is not in the cache, (nil, false) will be returned.
 func (c *ObjectLRU) Get(k plumbing.Hash) (plumbing.EncodedObject, bool) {
+	c.mut.Lock()
```

Comment:
`Get` mutates the state in `c.ll.MoveToFront(ee)`, that's why I used a `Mutex` instead of a `RWMutex`